contracts/implementations/streamr/VehicleStream.sol

```diff
@@ -372,6 +373,50 @@ contract VehicleStream is AccessControlInternal {
         }
     }
 
+    /**
+     * @notice Set a subscription permission for a user when
+     * the privilege to subscribe to live data is set in the VehicleId contract
+     * @dev Can only be called by the VehicleId contract
+     * @param vehicleId Vehicle node Id
+     * @param subscriber Vehicle stream subscriber
+     */
+    function onSetSubscribePrivilege(
```

Comment:
Did we have any prescribed revert behavior here?

Reply:
Shouldn't we grant the subscription for as long as the privilege lasts? Privileges have expirations too.

What if the expiration is 0 (we're un-granting, in some sense)?

Reply:
oh dear... fixing it.

I don't think we have revert cases here. Since it is a hook from the Vehicle Id, it would be weird some unexpected revert. This is why I just return if the registry does not have the grant permission.

Reply:
When we transfer a vehicle NFT, we wipe all of the privileges. Do we need logic somewhere to wipe the subscription?

Reply:
It is already covered. We call [onTransferVehicleStream](https://github.com/DIMO-Network/dimo-identity/blob/dev/contracts/NFTs/VehicleId.sol#L175C9-L175C60) in the VehicleId. And it deletes the stream [here](https://github.com/DIMO-Network/dimo-identity/blob/dev/contracts/implementations/streamr/VehicleStream.sol#L298) if it is official. This wipes the subscription.